### PR TITLE
Ignore permission validation for JWT authentication in DCR

### PR DESCRIPTION
**Pull request by ruks**
- Ignore permission validation for JWT authentication in DCR
- Fixes for https://github.com/wso2-enterprise/choreo/issues/15038

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java [hunk 1]
```diff
@@ -262,6 +262,12 @@ public Response unRegister(@QueryParam("applicationName") String applicationName
      * @return true if user has any of create, publish or subscribe permissions
      */
     private boolean isUserAccessAllowed(String username) {
+        String authScheme = (String) securityContext.get(RestApiConstants.REQUEST_AUTHENTICATION_SCHEME);
+        if (StringUtils.equals(authScheme, RestApiConstants.JWT_AUTHENTICATION)) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java:266]: lets add OPAQUE type too.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java [hunk 2]
```diff
@@ -262,6 +262,12 @@ public Response unRegister(@QueryParam("applicationName") String applicationName
      * @return true if user has any of create, publish or subscribe permissions
      */
     private boolean isUserAccessAllowed(String username) {
+        String authScheme = (String) securityContext.get(RestApiConstants.REQUEST_AUTHENTICATION_SCHEME);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java:265]: Let's add a comment mentioning, 
`The legacy permission validation (publish, subscribe, create) is skipped when the authentication is OAuth (Opaque or JWT). There's already a scope validation engaged there so no need of another permission check.`

Comment (ruks) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java:268]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java (hunk 1 above)
Comment (ruks) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java [hunk 3]
```diff
@@ -258,10 +258,20 @@ public Response unRegister(@QueryParam("applicationName") String applicationName
...
+     * OAuth (Opaque or JWT). There's already a scope validation engaged there so no need of another permission check.
+     *
      * @param username username
      * @return true if user has any of create, publish or subscribe permissions
      */
     private boolean isUserAccessAllowed(String username) {
+        String authScheme = (String) securityContext.get(RestApiConstants.REQUEST_AUTHENTICATION_SCHEME);
+        if (StringUtils.equals(authScheme, RestApiConstants.JWT_AUTHENTICATION) || StringUtils.equals(authScheme,
+                RestApiConstants.OPAQUE_AUTHENTICATION)) {
+            log.debug("Ignore permission validation as authentication schema is JWT");
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java:271]:
```suggestion
            log.debug("Ignore permission validation as authentication schema is OAuth2");
```

**Review by malinthaprasan — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java (hunk 3 above)
Comment (ruks) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java]: Fixed

**Pull request merged**